Sources/ArgumentParser/Completions/ZshCompletionsGenerator.swift

```diff
@@ -9,217 +9,236 @@
 //
 //===----------------------------------------------------------------------===//
 
-struct ZshCompletionsGenerator {
+extension [ParsableCommand.Type] {
   /// Generates a Zsh completion script for the given command.
-  static func generateCompletionScript(_ type: ParsableCommand.Type) -> String {
-    let initialFunctionName = [type].completionFunctionName()
+  var zshCompletionScript: String {
+    """
+    #compdef \(first?._commandName ?? "")
```

Comment:
It would be an internal error if `self` were empty here, so I'd rather fully fail than end up with an unusable script. Can you bind `first` before rendering the string, and then trap otherwise?

Reply:
Is a `fatalError()` in a `guard` block OK?

If so, is `first!` OK as long as I silence lint warnings with a `// swift-format-ignore: NeverForceUnwrap` comment?

I now notice that's used 24 times elsewhere under `Sources`.

Reply:
`first!` is OK with both the format-ignore comment and an explanatory comment (e.g. `// Precondition: self is guaranteed to be non-empty`).

Switching from `?? ""` to the force unwrap sounds good where it makes sense – if you need it multiple times in a function I'd rather you bind the unwrapped value to a local variable. Let's keep the format-ignore comments as local as possible.